internal/daemon/daemon.go

```diff
@@ -26,17 +26,25 @@ func port() int {
 	return port
 }
 
-func daemonRequest(endpoint string, body interface{}) (*http.Response, error) {
+func daemonRequest(endpoint string, body interface{}, method string) (*http.Response, error) {
```

Comment:
is there a possibility that this requires a default value? for example, POST since most calls are post calls.

just a suggestion nothing that would impede this PR to be approved as it looks gooooood!

Reply:
I'm glad you brought that up. Golang doesn't support optional parameters by design. Because of that all calls to this function must pass the method OR we split out another "SyncGetRequest" function to intentionally have a path for GET requests which doesn't make much sense.